applications/nrf_desktop/doc/hid_forward.rst

```diff
@@ -76,13 +78,18 @@ Forwarding HID input reports
 ============================
 
 After :ref:`nrf_desktop_ble_discovery` successfully discovers a connected peripheral, the |hid_forward| automatically subscribes for every HID input report provided by the peripheral.
+Reports in both HID report protocol and HID boot protocol can be used by the subscriber.
+However with current implementation, the USB state module supports either HID boot keyboard or HID boot mouse reports, because the boot report type is set as a single Kconfig option.
 
 The :c:func:`hidc_read` callback is called when HID input report is received from the connected peripheral.
 The received HID input report data is converted to ``hid_report_event``.
 
 ``hid_report_event`` is submitted and then the HID-class USB device configured by :ref:`nrf_desktop_usb_state` forwards it to the host.
 When a HID report is sent to the host by the HID-class USB device, the |hid_forward| receives a ``hid_report_sent_event`` with the identifier of this device.
 
+HID boot protocol reports are received with report ID equal REPORT_ID_RESERVED and it is later changed based on hogp objects by |hid_forward| to either REPORT_ID_BOOT_KEYBOARD or REPORT_ID_BOOT_MOUSE.
+While receiving subscription event, |hid_forward| updates report protocol in case if report protocol of the subscriber had changed.
```

Comment:
I would move it to interaction with USB, since we describe subscription mechanism there